geopandas/tools/clip.py

```diff
@@ -180,25 +180,53 @@ def clip(gdf, clip_obj, drop_slivers=False):
     order = pd.Series(range(len(gdf)), index=gdf.index)
     concat = pd.concat([point_gdf, line_gdf, poly_gdf])
 
-    if (concat.geom_type == "GeometryCollection").any() and drop_slivers:
-        concat = concat.explode()
-        if not polys.empty:
-            concat = concat.loc[concat.geom_type.isin(["Polygon", "MultiPolygon"])]
-        elif not lines.empty:
-            concat = concat.loc[
-                concat.geom_type.isin(["LineString", "MultiLineString", "LinearRing"])
-            ]
-        elif not points.empty:
-            concat = concat.loc[concat.geom_type.isin(["Point", "MultiPoint"])]
-        else:
-            raise TypeError("`drop_sliver` does not support {}.".format(type))
-    elif (concat.geom_type == "GeometryCollection").any() and not drop_slivers:
+    polys = ["Polygon", "MultiPolygon"]
+    lines = ["LineString", "MultiLineString", "LinearRing"]
+    points = ["Point", "MultiPoint"]
+
+    poly_check_orig = gdf.geom_type.isin(polys).any()
```

Comment:
These three variables are not needed if you just add the boolean values directly in the list below.